tests/base/develop_pipelines.py

```diff
@@ -119,3 +128,14 @@ def run_prediction(dataloader, trained_model, dp=False, min_acc=0.50):
         acc = acc.item()
 
     assert acc >= min_acc, f"This model is expected to get > {min_acc} in test set (it got {acc})"
+
+
+@run_prediction.register(BoringModel)
```

Comment:
I am still not very convinced about this wrapping, I see the beauty but it is not trivial for new users...

Reply:
Hi @borda and thanks for the review! I see your point, this could be hard to follow for someone new to python.Sadly, python doesn't have polimorphysm. If we don't want to add extra methods to `boring model`, the only way is to dispatch `run_predictions` calls here, distinguishing per classes and subclasses (a lot of tests inherit BoringModel). This is what`singledispatch` is for. Maybe i can add some comments to explain the approach, does it sound good to you? 
Otherwise, we need to use nested if for dispatching, but then the code loses in clarity (at least for me) . In any case no problem, let me know and I'll fix this :) 
Many thanks!

Reply:
lets see what others think... @PyTorchLightning/core-contributors 
for me, we shall keep the contribution barrier as low as possible